mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        scatter_spectra = np.load(
+        scatter_spectra_file_path, allow_pickle = True
+        )
+        en_array = self.std_eV_array()
+        current_full_spectrum = np.zeros(len(en_array))
+        emitted_peak = self.base_shape
+        if emitted_peak == 'lorentzian':
+            current_working_spectrum = self.std_lorenztian_17keV()
+        elif emitted_peak == 'shake':
+            current_working_spectrum = self.shakeSpectrumClassInstance.shake_spectrum()
+        elif emitted_peak == 'dirac':
+            current_working_spectrum = self.std_dirac()
+        current_working_spectrum = self.convolve_gaussian(current_working_spectrum, gauss_FWHM_eV)
+        zeroth_order_peak = current_working_spectrum
+        current_full_spectrum += current_working_spectrum
+        N = len(self.gases)
+        for M in range(1, self.max_scatters + 1):
+            gas_scatter_combinations = np.array([np.array(i) for i in product(range(M+1), repeat=N) if sum(i)==M])
+            for combination in gas_scatter_combinations:
+                entry_str = ''
+                for component, gas_type in zip(combination, self.gases):
+                    entry_str += gas_type
+                    entry_str += str(component).zfill(2)
+                current_working_spectrum = scatter_spectra.item()[entry_str]
+                current_working_spectrum = self.normalize(signal.convolve(zeroth_order_peak, current_working_spectrum, mode='same'))
+                coefficient = factorial(sum(combination))
+                for component, i in zip(combination, range(len(self.gases))):
+                    coefficient = coefficient/factorial(component)*p[i]**component
+                current_full_spectrum += coefficient*current_working_spectrum*prob_parameter**M
+        return current_full_spectrum
+
+    def spectrum_func_1(self, bins_Hz, *p0):
+        B_field = p0[0]
+        FWHM_G_eV = p0[1]
+        amplitude = p0[2]
+        prob_parameter = p0[3]
+
+        x_eV = ConversionFunctions.Energy(bins_Hz, B_field)
+        en_loss_array = self.std_eV_array()
+        en_loss_array_min = en_loss_array[0]
+        en_loss_array_max = en_loss_array[len(en_loss_array)-1]
+        en_array_rev = ComplexLineShapeUtilities.flip_array(-1*en_loss_array)
+        f = np.zeros(len(x_eV))
+        f_intermediate = np.zeros(len(x_eV))
+
+        x_eV_minus_line = Constants.kr_k_line_e() - x_eV
+        zero_idx = np.r_[np.where(x_eV_minus_line< en_loss_array_min)[0],np.where(x_eV_minus_line>en_loss_array_max)[0]]
+        nonzero_idx = [i for i in range(len(x_eV)) if i not in zero_idx]
+
+        full_spectrum = self.make_spectrum_1(FWHM_G_eV, prob_parameter,)
+        f_intermediate[nonzero_idx] = np.interp(x_eV_minus_line[nonzero_idx],en_array_rev,full_spectrum)
+        f[nonzero_idx] += amplitude*f_intermediate[nonzero_idx]/np.sum(f_intermediate[nonzero_idx])
+        return f
+
+    def fit_data_1(self, freq_bins, data_hist_freq):
```

Comment:
add description

Reply:
Added the description for the fit_data_1() function.